test/tap/adduser-always-auth.js

```diff
@@ -46,19 +63,37 @@ test('npm login', function (t) {
       })
     })
 
-    var o = ''
-    var e = ''
-    var remaining = Object.keys(responses).length
-    runner.stdout.on('data', function (chunk) {
-      remaining--
-      o += chunk
-
-      var label = chunk.toString('utf8').split(':')[0]
-      runner.stdin.write(responses[label])
+    var message = 'Logged in as u on ' + common.registry + '/.'
+    runner.stdout.on('data', verifyStdout(runner, message, t))
+  })
+})
 
-      if (remaining === 0) runner.stdin.end()
+test('npm login --scope', function (t) {
```

Comment:
@othiym23, maybe the `--scope` test should check a few other things as well? 
Examples:
* That `@scope:registry` is set to the registry URI in the config upon completion.
* That the correct registry URI is set when a scope is given.

I.e. this block of code: https://github.com/npm/npm/blob/master/lib/adduser.js#L133-L140 (by the way, what does the multiple argument `npm.config.get` do here (`npm.config.get('registry', 'cli')`)?

Could test the `@` prefixing of scopes too.